src/test/java/bio/terra/service/dataset/AssetModelValidatorTest.java

```diff
@@ -1,68 +1,88 @@
 import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
 import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
 
+import bio.terra.app.controller.ApiValidationExceptionHandler;
+import bio.terra.app.controller.DatasetsApiController;
 import bio.terra.common.TestUtils;
 import bio.terra.common.category.Unit;
-import bio.terra.common.fixtures.JsonLoader;
+import bio.terra.common.iam.AuthenticatedUserRequestFactory;
 import bio.terra.model.ErrorModel;
+import bio.terra.service.auth.iam.IamService;
+import bio.terra.service.filedata.FileService;
+import bio.terra.service.job.JobService;
+import bio.terra.service.snapshotbuilder.SnapshotBuilderService;
 import java.util.UUID;
-import org.apache.commons.lang3.StringUtils;
-import org.junit.Test;
-import org.junit.experimental.categories.Category;
-import org.junit.runner.RunWith;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
-import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
 import org.springframework.http.MediaType;
-import org.springframework.mock.web.MockHttpServletResponse;
 import org.springframework.test.context.ActiveProfiles;
-import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.context.ContextConfiguration;
 import org.springframework.test.web.servlet.MockMvc;
-import org.springframework.test.web.servlet.MvcResult;
 
-@RunWith(SpringRunner.class)
-@SpringBootTest(properties = {"datarepo.testWithEmbeddedDatabase=false"})
-@AutoConfigureMockMvc
 @ActiveProfiles({"google", "unittest"})
-@Category(Unit.class)
-public class AssetModelValidatorTest {
+@ContextConfiguration(
+    classes = {
+      AssetModelValidator.class,
+      DatasetsApiController.class,
+      ApiValidationExceptionHandler.class
+    })
+@WebMvcTest
+@Tag(Unit.TAG)
+class AssetModelValidatorTest {
+  @MockBean private JobService jobService;
+  @MockBean private DatasetService datasetService;
+  @MockBean private IamService iamService;
+  @MockBean private FileService fileService;
+  @MockBean private AuthenticatedUserRequestFactory authenticatedUserRequestFactory;
+  @MockBean private SnapshotBuilderService snapshotBuilderService;
+  @MockBean private IngestRequestValidator ingestRequestValidator;
+  @MockBean private DataDeletionRequestValidator dataDeletionRequestValidator;
+  @MockBean private DatasetSchemaUpdateValidator datasetSchemaUpdateValidator;
+  @MockBean private DatasetRequestValidator datasetRequestValidator;
 
   @Autowired private MockMvc mvc;
-  @Autowired private JsonLoader jsonLoader;
+
+  @BeforeEach
+  void beforeEach() {
```

Comment:
The validator tests require similar boilerplate, it would be nice if there was a way to share this code.

Reply:
I often find that stubs in before each methods trigger strict stubbing errors, but that seems to not be a problem here

Reply:
The difference here isn't that they're `@BeforeEach`, it's that using the spring boot test annotations doesn't support strict stubbing. As far as I can tell, there's no easy way to enable strict stubbing and also use `WebMvcTest`. See https://github.com/spring-projects/spring-boot/issues/19383. Although I do see a SO response that could work https://stackoverflow.com/a/75388020/947286